cds_dojson/marc21/fields/books/base.py

```diff
@@ -432,22 +433,50 @@ def get_value_rn(f_a, f_z, f_9, rn_obj):
     sub_a = clean_val('a', value, str)
     sub_z = clean_val('z', value, str)
 
-    if not (sub_z or sub_a or sub_9):
-        raise MissingRequiredField(subfield='9 or a or z')
+    all_empty = not (sub_z or sub_a or sub_9)
 
     if key == '037__':
+        if all_empty:
+            raise MissingRequiredField(subfield='9 or a or z')
+
         if sub_9 == 'arXiv':
             self['arxiv_eprints'] = arxiv_eprints(self, key, value)
             raise IgnoreKey('report_numbers')
         else:
             get_value_rn(sub_a, sub_z, sub_9, f)
 
     if key == '088__':
+        if 'n' in value or 'x' in value:
+            barcodes(self, key, value)
+
+        if all_empty and 'n' not in value and 'x' not in value:
+            raise MissingRequiredField(subfield='9 or a or z or n or x')
+
         get_value_rn(sub_a, sub_z, sub_9, f)
 
     return f
 
 
+@model.over('barcodes', '^088__')
+@out_strip
+@for_each_value
+def barcodes(self, key, value):
+    """Match barcodes to volumes."""
+    val_n = clean_val('n', value, str)
+    val_x = clean_val('x', value, str)
+
+    _migration = self.get('_migration', {'volumes': []})
+    _migration['volumes'].append(dict(
+        volume=extract_volume_number(
+            val_n,
+            raise_exception=True,
+            subfield='n'
+        ),
+        barcode=val_x
+    ))
+    self['_migration'] = _migration
+
```

Comment:
just for clarity, you should put here the `IgnoreKey` for `barcodes` because right now it returns none and `@out_strip` cleans it, `Ignore Key` would be better

Reply:
`@out_strip` is now removed and also raising `IgnoreKey`.